src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,58 +1,127 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    if (!title || !imgUrl || !imdbUrl || !imdbId) {
+      return;
+    }
+
+    setCount(count + 1);
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+  };
+
+  const reset = () => {
+    setTitle('');
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbId('');
+  };
+
+  function SubmitButton() {
+    if (title && imgUrl && imdbUrl && imdbId) {
+      return (
+        <div className="field is-grouped">
+          <div className="control">
+            <button
+              type="submit"
+              data-cy="submit-button"
+              className="button is-link"
+            >
+              Add
+            </button>
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <div className="field is-grouped">
+        <div className="control">
+          <button
+            type="submit"
+            data-cy="submit-button"
+            className="button is-link"
+            disabled
+          >
+            Add
+          </button>
+        </div>
+      </div>
+    );
```

Comment:
do not return 2 similar code snippets
```suggestion
    const isFormFilled = title && imgUrl && imdbUrl && imdbId;

    return (
      <div className="field is-grouped">
        <div className="control">
          <button
            type="submit"
            data-cy="submit-button"
            className="button is-link"
            disabled={!isFormFilled}
          >
            Add
          </button>
        </div>
      </div>
    );
```